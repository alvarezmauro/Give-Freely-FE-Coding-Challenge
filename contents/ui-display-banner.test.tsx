import { render, screen } from "@testing-library/react";

import Banner from "./ui-display-banner";

// Data we expect to be returned from getParticipantsData
export const MOCK_PARTICIPANTS_DATA = {
  websites: [
    {
      name: "TripAdvisor",
      url: "www.tripadvisor.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    },
    {
      name: "Uber",
      url: "www.uber.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    },
    {
      name: "Amazon",
      url: "www.amazon.com",
      messages: ["Message 1", "Message 2", "Message 3", "Message 4"]
    }
  ]
};

// Mock @plasmohq/storage/hook
jest.mock("@plasmohq/storage/hook", () => ({
  useStorage: () => [MOCK_PARTICIPANTS_DATA]
}));

describe("ui-display-banner", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render the banner when we are in a participating website", async () => {
    // mock window.location.host
    delete window.location;
    window.location = {
      host: "www.tripadvisor.com"
    } as any;

    render(<Banner />);
    const renderedBanner = await screen.queryByTestId("banner");
    expect(renderedBanner).toBeInTheDocument();
  });

  it("should not render the banner when we aren't in a participating website", async () => {
    // mock window.location.host
    delete window.location;
    window.location = {
      host: "www.google.com"
    } as any;

    render(<Banner />);
    const renderedBanner = await screen.queryByTestId("banner");
    expect(renderedBanner).not.toBeInTheDocument();
  });
});
