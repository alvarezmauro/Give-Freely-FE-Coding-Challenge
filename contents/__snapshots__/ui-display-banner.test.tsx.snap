// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-display-banner Snapshot test 1`] = `
<div
  data-testid="banner"
  style="background-color: rgba(255, 255, 255, 0.8); width: 100%; position: fixed;"
>
  <p
    style="text-align: center;"
  >
    Message 1
  </p>
</div>
`;
